### PR TITLE
Evolutionary based black-box adversarial attack.

### PR DESCRIPTION
**Pull request by aliotopal**
# Description

I have developed an evolutionary algorithm-based black-box adversarial attack (attack_EA). It is added in evasion attacks section.

Fixes # (issue)

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] Test A
- [x] Test B

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by aliotopal** [comment]
Signed-off-by: Ali Osman TOPAL <aliotopal@gmail.com>

**Comment by beat-buesser** [comment]
Hi @aliotopal Thank you for your pull request! I'll change the target branch to the dev-branch for next release 1.16.

**Comment by beat-buesser** [comment]
Hi @aliotopal Could you please take a look at the Style Checks? You can solve most of the issues by running Black formatter version `black==21.12b0` with commands `black --line-length 120 art/` and `black --line-length 120 tests/`.

**Comment by aliotopal** [comment]
Hi, I have done the style check, it should be ok now. Can you please have a look if everything is ok? Best,

**Comment by aliotopal** [comment]
Hi @beat-buesser,
We have updated the code so it is compatible with ART libraries and the attack is working now. It is written in Keras only. Can you please have a look when you have time?

**Comment by aliotopal** [comment]
Hi @beat-buesser, I went over the style errors, it should be fine now (I hope). 
Style Check:

1. Module name "attack_EA" doesn't conform to snake_case naming style (invalid-name): changed name to attack_ea
2. Variable name "x_" doesn't conform to snake_case naming style (invalid-name): x_ changed to x_temp
3. Argument 1 to "shuffle" has incompatible type "ndarray[Any, Any]"; expected "MutableSequence[Any]: removed function, it was redundant.
4. generate function arguments are changed according to the superclass.

For the other errors i don't know what to do.

can you please have a look once more?

**Comment by aliotopal** [comment]
Dear @beat-buesser,

I am sorry, I know I am becoming a pain :(, but when you have time can you please have a look at our attack?

**Comment by aliotopal** [comment]
Dear @beat-buesser,

Finally, style check is giving only this error: "Class name "attack_ea" doesn't conform to PascalCase naming style (invalid-name)" I don't want to change file name, if it is ok. Besides that, there are some more errors in "mypy" but it seems they are not related to our attack. Can you please have a look and give us direction?

**Comment by aliotopal** [comment]
Dear @beat-buesser,

Finally, style check is giving only this error: "Class name "attack_ea" doesn't conform to PascalCase naming style (invalid-name)" I don't want to change file name, if it is ok. Besides that, there are some more errors in "mypy" but it seems they are not related to our attack. Can you please have a look and give us direction?

________________________________

**Comment by aliotopal** [comment]
Hi @beat-buesser<https://github.com/beat-buesser>,

I cleared the Style Checks error. There are some error messages from mypy: warpPerspective which has nothing to do with our attack as far as I understand.

In CI Keras test: 4 jobs failed, but the error is Error uploading to https://codecov.io<https://codecov.io/>

there is nothing that I can do I guess.

Please let me know if there is something that I can do.

Best,
Ali

________________________________
